cns/middlewares/mock/mockClient.go

```diff
@@ -42,25 +42,75 @@ func NewClient() *Client {
 	testPod4.Labels = make(map[string]string)
 	testPod4.Labels[configuration.LabelPodSwiftV2] = podNetwork
 
+	testPod5 := v1.Pod{}
```

Comment:
just a side note, but I recently learned about this package https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.18.3/pkg/client/fake, in hindsight this would make this whole process of mocking up a client a lot cleaner, instead of having to do all of this 🥲 might be out of scope for this PR but FYI

Reply:
Do we need to apply this to our new tests?

Reply:
no, this would require us to rework the entire current test implementation, which would be out of scope for this PR, just something that I thought of 🙂